### PR TITLE
[Android] Add preview support for compose

### PR DESCRIPTION
**Pull request by jonnyandrew**
## Changes

- Allow state to be created with an initial value
- Add preview to the example app
- Support compose preview by replacing the rich text editor with a plain EditText when in preview mode

## Context

- Part of https://github.com/matrix-org/matrix-rich-text-editor/issues/315

### CODE REVIEW

File: platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt [hunk 1]
```diff
@@ -106,12 +120,51 @@ fun RichTextEditor(
...
+    if (!LocalInspectionMode.current) {
+        throw IllegalStateException("PreviewEditor should only be used in preview mode")
+    }
+
+    val context = LocalContext.current
+
+    AndroidView(
+        modifier = modifier,
+        factory = {
+            val view = AppCompatEditText(context).apply {
```
Comment (julioromano) [inline platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt:143]: Why not just a material `TextField` ? It'd be a one-liner

File: platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditorState.kt [hunk 2]
```diff
@@ -22,7 +22,19 @@ import uniffi.wysiwyg_composer.MenuAction
...
  */
 @Stable
-class RichTextEditorState internal constructor() {
+class RichTextEditorState internal constructor(
+    initialHtml: String = "",
+) {
+    /**
+     * The content of the editor as HTML formatted for sending as a message.
+     */
+    var messageHtml by mutableStateOf("")
```
Comment (julioromano) [inline platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditorState.kt:31]: `var messageHtml by mutableStateOf(initialHtml)` ? (removing the `init{}` block)

**Review by julioromano — APPROVED** [review]
Just a few suggestion, but feel free to ignore them

File: platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt (hunk 1 above)
Comment (jonnyandrew) [reply platforms/android/library-compose/src/main/java/io/element/android/wysiwyg/compose/RichTextEditor.kt:143]: I wanted the preview editor to match the real editor as closely as possible. Using an `EditText` for the preview allows us to make changes to the real editor that will automatically get reflected in the preview, whereas using `TextField` would make it harder to get it looking pixel perfect.

**Pull request merged**